services/web/server/src/simcore_service_webserver/login/_2fa_handlers.py

```diff
@@ -0,0 +1,145 @@
+import logging
+from typing import Literal
+
+from aiohttp import web
+from aiohttp.web import RouteTableDef
+from models_library.emails import LowerCaseEmailStr
+from pydantic import Field
+from servicelib.aiohttp import status
+from servicelib.aiohttp.requests_validation import parse_request_body_as
+from servicelib.mimetype_constants import MIMETYPE_APPLICATION_JSON
+
+from ..products.api import Product, get_current_product
+from ..session.access_policies import session_access_required
+from ._2fa_api import (
+    create_2fa_code,
+    delete_2fa_code,
+    get_2fa_code,
+    mask_phone_number,
+    send_email_code,
+    send_sms_code,
+)
+from ._constants import (
+    CODE_2FA_EMAIL_CODE_REQUIRED,
+    CODE_2FA_SMS_CODE_REQUIRED,
+    MSG_2FA_CODE_SENT,
+    MSG_EMAIL_SENT,
+    MSG_UNKNOWN_EMAIL,
+)
+from ._models import InputSchema
+from .errors import handle_login_exceptions
+from .settings import LoginSettingsForProduct, get_plugin_settings
+from .storage import AsyncpgStorage, get_plugin_storage
+from .utils import envelope_response
+
+_logger = logging.getLogger(__name__)
+
+
+routes = RouteTableDef()
+
+
+class Resend2faBody(InputSchema):
+    email: LowerCaseEmailStr = Field(..., description="User email (identifier)")
+    via: Literal["SMS", "Email"] = "SMS"
+
+
+@routes.post("/v0/auth/two_factor:resend", name="auth_resend_2fa_code")
+@session_access_required(
+    name="auth_resend_2fa_code",
+    one_time_access=False,
+)
+@handle_login_exceptions
+async def resend_2fa_code(request: web.Request):
+    """Resends 2FA code via SMS/Email"""
+    product: Product = get_current_product(request)
+    settings: LoginSettingsForProduct = get_plugin_settings(
+        request.app, product_name=product.name
+    )
+    db: AsyncpgStorage = get_plugin_storage(request.app)
+    resend_2fa_ = await parse_request_body_as(Resend2faBody, request)
+
+    user = await db.get_user({"email": resend_2fa_.email})
+    if not user:
+        raise web.HTTPUnauthorized(
```

Comment:
There seem to be implicit assumptions here, should this not return unauthorized but actually 404 (*dont* use 404 though) or better `GONE`

Reply:
This is not my code, just refactored/improved it a bit but I didn't modify the interface (Not sure whether it would cause some troubles to the front end if I changed it.).